spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/write/BigQueryWriteHelper.java

```diff
@@ -105,7 +124,23 @@ public void writeDataFrameToBigQuery() {
       String format = config.getIntermediateFormat().getDataSource();
       data.write().format(format).save(gcsPath.toString());
 
-      loadDataToBigQuery();
+      if (writeDisposition == JobInfo.WriteDisposition.WRITE_TRUNCATE
+          && config.getPartitionOverwriteModeValue() == PartitionOverwriteMode.DYNAMIC
+          && bigQueryClient.tableExists(config.getTableId())) {
+        temporaryTableId = createTempTable(config.getTableId(), tableSchema);
+        loadDataToBigQuery();
+        Job queryJob =
+            bigQueryClient.overwriteDestinationWithTemporaryDynamicPartitons(
+                temporaryTableId, config.getTableId());
+        BigQueryClient.waitForJob(queryJob);
+        Preconditions.checkState(
```

Comment:
The table deletion should be initiated in the `BigQueryClient.createTempTable()` method, you can add the snippet 
```
      // Registering a cleanup job
      CLEANUP_JOBS.add(() -> deleteTable(tempTable));
```

Reply:
Done